OPHD/States/MapViewStateHelper.cpp

```diff
@@ -559,11 +561,34 @@ NAS2D::Xml::XmlElement* writeRobots(RobotPool& robotPool, RobotTileTable& robotM
 }
 
 
-NAS2D::Xml::XmlElement* writeResearch(const ResearchTracker&)
+NAS2D::Xml::XmlElement* writeResearch(const ResearchTracker& tracker)
 {
 	auto* research = new NAS2D::Xml::XmlElement("research");
 
+	std::string completedResearch;
+	for (auto techId : tracker.completedResearch())
+	{
+		completedResearch += std::to_string(techId) + ",";
+	}
```

Comment:
```cpp
const auto completedResearch = NAS2D::join(tracker.completedResearch(), ",");
```

Reply:
Was just thinking `int` needed to be converted to `std::string` before the call to `join`:
```cpp
NAS2D::mapToVector(tracker.completedResearch(), [](const auto& x){return std::to_string(x);})
```

All together, that would be:
```cpp
const auto completedResearch = NAS2D::join(NAS2D::mapToVector(tracker.completedResearch(), [](const auto& x){return std::to_string(x);}), ",");
```

Reply:
Ideally, `mapToVector` should work with a function reference, though I've noticed the compiler fails to deduce the template parameter in that case. Perhaps I need to look into this. Ideally it should be possible to convert using:
```cpp
NAS2D::mapToVector(tracker.completedResearch(), std::to_string); // Doesn't currently work, due to failed template deduction
```